### PR TITLE
Feature/realsense

### LINKED ISSUE #14: Realsense D435/T265 Modules
Create modules hardware abstraction modules for Intel Realsense D435 and T265 modules. [Previous Work](https://github.com/fsherratt/ERL_SmartCities_2019/tree/master/modules/realsense)

### PR DESCRIPTION
**Pull request by fsherratt**
Improvements to the structure of realsense objects.

- Created a parent class to contain core API interactions
- Split up Depth and Color streams
- Prepared code for integration with Rabbit MQ
- Removed depth post-processing code - to be added to a new node

Closes #14

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -38,7 +38,7 @@ RUN apt-get install -y python3 python3-pip python3-dev \
   && cd /usr/local/bin \
   && pip3 install --upgrade pip
 
-RUN pip3 install numpy pylint scipy coverage pytest
+RUN pip3 install numpy pylint scipy coverage pytest OmegaConf
```
Comment (fsherratt) [inline Dockerfile:41]: This was to fix the load_config function

File: tests/test_load_config.py [hunk 2]
```diff
@@ -22,7 +22,7 @@ def test_cli_input(self):
         sys.argv = [__name__, 'test={}'.format(self.test_value)] # Mock CLI inpuut
         conf = load_config()
         
-        self.assertEquals(conf.test, self.test_value)
+        self.assertEqual(conf.test, self.test_value)
```
Comment (fsherratt) [inline tests/test_load_config.py:25]: assertEquals is being deprecated

**Comment by fsherratt** [comment]
Only ↓ need a proper review the other files are just making pylint be quiet.
- docs/Publisher_List.md 
- docs/realsense.md 
- modules/realsense.py 
- tests/test_realsense.py

_n.b. Realsense code has been condensed down into a single file_

File: modules/realsense.py [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+                                   .format(self._object_name))
+            raise raised_exception
+
+        self._post_connect_process()
+
+        print('rs_pipeline:{}: Connection Open'.format(self._object_name))
+
+    def close_connection(self):
+        """
+        Close connection to D435 camera
```
Comment (Neoxas) [inline modules/realsense.py]: Comment should be generic

File: modules/realsense.py [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+        """
+        if traceback:
+            print(traceback.tb_frame)
+            self._exception_handle("rs_d435: __exit__: `{}`".format(exception_value))
+
+        self.close_connection()
+
+    def open_connection(self):
+        """
+        Open Connection to D435 camera
```
Comment (Neoxas) [inline modules/realsense.py]: Comment should be generic

File: modules/realsense.py [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+        """
+        Convert T265 rotational frame to aero NED frame
+        """
+        rot = self.h_aeroref_t265ref * R.from_quat(quat)  * self.h_t265body_aerobody
+
+        return rot.as_quat()
+
+    def _convert_positional_frame(self, pos) -> list:
+        """
+        Convert T264 translation frame to aero NED translation
```
Comment (Neoxas) [inline modules/realsense.py]: *T265

**Review by Neoxas — APPROVED** [review]

**Pull request merged**